backend/tests/unit/integrations/llm/test_llm_client.py

```
@@ -62,16 +64,35 @@ def test_initialization_xai_success(self, mock_config):
         with patch('tarsy.integrations.llm.client.ChatXAI') as mock_xai:
             mock_xai.return_value = Mock()
             
-            client = LLMClient("grok", mock_config)
+            # Use 'xai' as provider name and 'type': 'xai' in config
+            mock_config = {**mock_config, "type": "xai"}
+            client = LLMClient("xai", mock_config)
             
-            assert client.provider_name == "grok"
+            assert client.provider_name == "xai"
             assert client.available == True
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `client.available:` for truth checks

Replace with `client.available`

(E712)